debezium-core/src/main/java/io/debezium/transforms/tracing/ActivateTracingSpan.java

```diff
@@ -112,93 +97,35 @@ public void setRequireContextField(boolean requireContextField) {
     }
 
     @Override
-    public R apply(R record) {
+    public R apply(R connectRecord) {
         // In case of tombstones or non-CDC events (heartbeats, schema change events),
         // leave the value as-is
-        if (record.value() == null || !smtManager.isValidEnvelope(record)) {
-            return record;
+        if (connectRecord.value() == null || !smtManager.isValidEnvelope(connectRecord)) {
+            return connectRecord;
         }
 
-        final Struct envelope = (Struct) record.value();
+        final Struct envelope = (Struct) connectRecord.value();
 
         final Struct after = (envelope.schema().field(Envelope.FieldName.AFTER) != null) ? envelope.getStruct(Envelope.FieldName.AFTER) : null;
         final Struct source = (envelope.schema().field(Envelope.FieldName.SOURCE) != null) ? envelope.getStruct(Envelope.FieldName.SOURCE) : null;
         String propagatedSpanContext = null;
 
-        if (after != null) {
-            if (after.schema().field(spanContextField) != null) {
-                propagatedSpanContext = after.getString(spanContextField);
-            }
+        if (after != null && after.schema().field(spanContextField) != null) {
+            propagatedSpanContext = after.getString(spanContextField);
         }
 
         if (propagatedSpanContext == null && requireContextField) {
-            return record;
+            return connectRecord;
         }
 
         try {
-            return traceRecord(record, envelope, source, after, propagatedSpanContext);
+            return ActivateTracingSpanDelegate.Builder.build().traceRecord(connectRecord, envelope, source, propagatedSpanContext, operationName);
```

Comment:
Hello, why it is needed to create new instance of `ActivateTracingSpanDelegate` for every message? It is performance penalty (mainly GC stuff) for no obvious reason. `ActivateTracingSpanDelegate` does not save its state or anything like that.

Reply:
You are right. I changed the instance method to static method. I took the opportunity to make other minor performance improvements.